### DISCUSSION TITLE
Thank you!

### DISCUSSION THREAD

**Discussion by LilyFirefly** [BurntSushi/bstr#147] [General]
Thank you for writing this library. The [`Finder` struct](https://docs.rs/bstr/latest/bstr/struct.Finder.html) was exactly what I needed to optimise my code today.

**Comment by BurntSushi** [comment]
No problem! Indeed, if you're doing a lot of substring searches on small haystacks where the needle remains invariant, avoiding reconstructing the searcher itself every time can be a huge savings!

**Reply by LilyFirefly** [reply]
Yeah, that was exactly my use-case. :smile:
